### PR TITLE
React list of goods (Hooks)

### PR DESCRIPTION
**Pull request by yuran007**
[DEMO LINK](https:yuran007.github.io/react_list-of-goods-hooks/)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # React list of goods (Hooks)
-- Replace `<your_account>` with your Github username in the [DEMO LINK](https://<your_account>.github.io/react_list-of-goods-hooks/)
+- Replace `<your_account>` with your Github username in the [DEMO LINK](https:yuran007.github.io/react_list-of-goods-hooks/)
```
Comment (aliceblunt8) [inline README.md:2]:
```suggestion
- Replace `<your_account>` with your Github username in the [DEMO LINK](https://yuran007.github.io/react_list-of-goods-hooks/)
```

File: src/App.tsx [hunk 2]
```diff
@@ -14,11 +30,54 @@ const goodsFromServer: string[] = [
...
+
+    nextList.reverse();
+    setList(nextList);
+  };
+
+  const handlerSortByAlphabet = () => {
+    const nextList = [...list];
+
+    nextList.sort((a, b) => compareGoods(a, b, 'alphabet'));
+    setList(nextList);
```
Comment (aliceblunt8) [inline src/App.tsx:51]: do this inside setList, use prevState

File: src/App.tsx [hunk 3]
```diff
@@ -14,11 +30,54 @@ const goodsFromServer: string[] = [
...
+  };
+
+  const handlerSortByLength = () => {
+    const nextList = [...list];
+
+    nextList.sort((a, b) => compareGoods(a, b, 'length'));
+    setList(nextList);
```
Comment (aliceblunt8) [inline src/App.tsx:58]: do this inside setList, use prevState

File: src/App.tsx [hunk 4]
```diff
@@ -14,11 +30,54 @@ const goodsFromServer: string[] = [
...
+    goodsFromServer,
+  );
+
+  const [clicked, setClicked] = useState(false);
+
+  const handlerToggleReverse = () => {
+    const nextList = [...list];
+
+    nextList.reverse();
+    setList(nextList);
```
Comment (aliceblunt8) [inline src/App.tsx:44]: do this inside setList, use prevState

File: src/components/List.tsx [hunk 5]
```diff
@@ -0,0 +1,9 @@
+type Props = {
+  list: string[];
+};
+
+export const List: React.FC<Props> = ({ list }) => {
+  const listItems = list.map(item => <li key={item}>{item}</li>);
+
+  return <ul>{listItems}</ul>;
```
Comment (aliceblunt8) [inline src/components/List.tsx:8]: no need for variable, just add inside ul

**Review by aliceblunt8 — CHANGES_REQUESTED** [review]

File: src/components/List.tsx (hunk 5 above)
Comment (yuran007) [reply src/components/List.tsx:8]: done

File: src/App.tsx (hunk 4 above)
Comment (yuran007) [reply src/App.tsx:44]: done

File: src/App.tsx (hunk 2 above)
Comment (yuran007) [reply src/App.tsx:51]: done

File: src/App.tsx (hunk 3 above)
Comment (yuran007) [reply src/App.tsx:58]: done

**Pull request closed without merging**